examples/versatileab/src/bin/abt-exception.rs

```diff
@@ -0,0 +1,77 @@
+//! Example triggering an data abort exception.
+
+#![no_std]
+#![no_main]
+
+use cortex_ar::register::{Dfar, Dfsr, Sctlr};
+// pull in our start-up code
+use versatileab as _;
+
+use semihosting::println;
+
+versatileab::entry_point!();
+
+/// The main function of our Rust application.
+#[export_name = "main"]
+#[allow(unreachable_code)]
+fn main() -> ! {
+    // Enable alignment check for Armv7-R. Was not required
+    // on Cortex-A for some reason, even though the bit was not set.
+    enable_alignment_check();
+
+    println!("Hello, this is an data abort exception example");
+    // Unaligned read
+    unsafe {
+        let addr: *const u32 = 0x1001 as *const u32; // Unaligned address (not 4-byte aligned)
+        core::arch::asm!(
+            "ldr r0, [{addr}]",  // Attempt unaligned load (should trigger Data Abort)
+            addr = in(reg) addr, // Pass unaligned pointer
+            options(nostack, preserves_flags) // No stack usage, preserves flags
+        );
+    }
+
+    unreachable!("should never be here!");
+}
+
+fn enable_alignment_check() {
+    let mut sctrl = Sctlr::read();
+    sctrl.set_a(true);
+    Sctlr::write(sctrl);
+}
+
+fn disable_alignment_check() {
+    let mut sctrl = Sctlr::read();
+    sctrl.set_a(false);
+    Sctlr::write(sctrl);
+}
+
+#[unsafe(no_mangle)]
+unsafe extern "C" fn _undefined_handler(_faulting_instruction: u32) {
+    panic!("unexpected undefined exception");
+}
+
+#[unsafe(no_mangle)]
+unsafe extern "C" fn _prefetch_handler(_faulting_instruction: u32) {
+    panic!("unexpected prefetch exception");
+}
+
+#[unsafe(no_mangle)]
+unsafe extern "C" fn _abort_handler(_faulting_instruction: u32) {
+    static mut COUNTER: u32 = 0;
```

Comment:
Re-entering this function using a static mut is UB. It should probably be at AtomicU32 using fetch_add.

Reply:
Right, replaced those with Atomics.